src/blocks/map/edit.js

```diff
@@ -0,0 +1,384 @@
+					markerDiv.className = 'marker';
+					markerDiv.dataset.id = id;
+
+					const pin = new mapboxgl.Marker( markerDiv )
+						.setDraggable( true )
+						.setLngLat( coords )
+						.addTo( map );
+
+					// If the item being added is the active item.
+					if ( index === mapItemIndex ) {
+						markerDiv.classList.add( 'active' );
+					}
+
+					// On click select the related slide and highlight the marker.
+					markerDiv.addEventListener( 'click', () => {
+						Array.from( mapMarkers ).forEach( ( div ) =>
+							div.classList.remove( 'active' )
+						);
+
+						markerDiv.classList.add( 'active' );
+
+						mapItemIndex = index;
+						setCurrentItemIndex( index );
+						selectBlock( blockId );
+					} );
+
+					// On marker drag, update its attributes.
+					pin.on( 'dragend', () => {
+						const lngLat = pin.getLngLat();
+						updateBlockAttributes( blockId, {
+							lat: lngLat.lat,
+							long: lngLat.lng,
+						} );
+					} );
+				}
+
+				// Push it to our list of existing markers.
+				if ( ! newMarkers.includes( id ) ) {
+					newMarkers.push( id );
+				}
+			}
+		}
+
+		// If there are markers which are on the map, but shouldn't be, remove them.
+		Array.from( mapMarkers ).forEach( ( mapMarker ) => {
+			if ( ! newMarkers.includes( parseInt( mapMarker?.dataset?.id ) ) ) {
+				mapMarker.remove();
+			}
+		} );
+
+		// If there are clusters which are on the map, but shouldn't be, remove them.
+		Array.from( clusterMarkers ).forEach( ( clusterMarker ) => {
+			if (
+				! newClusters.includes(
+					parseInt( clusterMarker?.dataset?.clusterId )
+				)
+			) {
+				clusterMarker.remove();
+			}
+		} );
+	}, [ selectBlock, updateBlockAttributes ] );
+
+	/**
+	 * If a slide is added, switch to the new slide. If one is deleted, make sure we don't
+	 * show a non-existent slide.
+	 */
+	useEffect( () => {
+		if ( slideBlocks.length > slideCount.current ) {
+			// Slide added
+			setCurrentItemIndex( slideBlocks.length - 1 );
+		} else if ( slideBlocks.length < slideCount.current ) {
+			// Slide deleted
+			if ( currentItemIndex + 1 > slideBlocks.length ) {
+				setCurrentItemIndex( slideBlocks.length - 1 );
+			}
+		}
+
+		// Update ref with new value..
+		slideCount.current = slideBlocks.length;
+	}, [ slideBlocks.length, currentItemIndex, slideCount, slideBlocks ] );
+
+	/**
+	 * Init the map.
+	 */
+	useEffect( () => {
+		// eslint-disable-next-line no-undef
+		if ( ! wmf?.apiKey ) {
+			return;
+		}
+
+		const fullScreenControl = new mapboxgl.NavigationControl();
+
+		if ( map ) {
+			map.removeControl( fullScreenControl );
+			map.remove();
+		}
+
+		// eslint-disable-next-line no-undef
+		mapboxgl.accessToken = wmf?.apiKey;
+		map = new mapboxgl.Map( {
+			container: 'map',
+			center: [ 2.8, 40 ],
+			projection: 'equalEarth',
+			scrollZoom: false,
+			style: mapStyle || 'mapbox://styles/mapbox/light-v11', // 'mapbox://styles/mattwatsonhm/clu09j0hw00tf01p7dpw5hyv7' >- custom grey colours.
```

Comment:
Is the commented out style something you're still working on, or can it be removed?